### PR TITLE
Add Network and Subnet resources for edgenetwork

### PR DESCRIPTION
**Pull request by f1urps**
This PR adds two new resources, `Network` and `Subnet`, under a new product `edgenetwork`.

Fixes this issue: https://github.com/hashicorp/terraform-provider-google/issues/15664

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
edgenetwork : `google_edgenetwork_network`
```

```release-note:new-resource
edgenetwork : `google_edgenetwork_subnet`
```

### CODE REVIEW

File: mmv1/products/edgenetwork/Subnet.yaml [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+      The ranges of ipv4 addresses that are owned by this subnetwork, in CIDR format.
+  - !ruby/object:Api::Type::Array
+    name: 'ipv6Cidr'
+    item_type: Api::Type::String
+    description: |
+      The ranges of ipv6 addresses that are owned by this subnetwork, in CIDR format.
+  - !ruby/object:Api::Type::Integer
+    name: 'vlanId'
+    default_from_api: true
+    send_empty_value: false
```
Comment (zli82016) [inline mmv1/products/edgenetwork/Subnet.yaml:124]: What is the reason to add send_empty_value: false here?

**Comment by zli82016** [comment]
Can you please add the labels field to the tests? Please check  the missing test report for details.

File: mmv1/products/edgenetwork/Subnet.yaml (hunk 1 above)
Comment (f1urps) [reply mmv1/products/edgenetwork/Subnet.yaml:124]: Is it false by default? I will remove that line if so.

Comment (zli82016) [reply mmv1/products/edgenetwork/Subnet.yaml:124]: Yeah, false is by default.

**Comment by f1urps** [comment]
> Can you also fix the failed tests? Thanks.

These tests are failing because they don't have a valid GDCE zone to operate on. Unfortunately the edgenetwork API does not support the creation of a zone without an actual hardware deployment. Can we skip these tests?

**Comment by zli82016** [comment]
> 

Is there a doc how to create a zone with an actual hardware deployment? Thanks.

**Comment by f1urps** [comment]
> Is there a doc how to create a zone with an actual hardware deployment? Thanks.

Internally in GDCE we have several zones created for testing/development purposes, which I can use to run these acceptance tests. However I'm not sure it's a good idea to expose the details for them in a public GitHub repo. Alternatively, a GDCE zone can be purchased and provisioned via [these steps](https://cloud.google.com/distributed-cloud/edge/latest/docs/order).

Currently it is not possible to run any kind of E2E test for the edgenetwork API without owning some GDCE hardware, either provided by Google internally or purchased as a customer. To me, neither of these options seem to make sense.

I can confirm these tests are passing when I run them on a test zone. Would it be okay to disable these tests in the CI, and leave them as manual-run only?

File: mmv1/products/edgenetwork/Subnet.yaml (hunk 1 above)
Comment (f1urps) [reply mmv1/products/edgenetwork/Subnet.yaml]: Done

**Comment by zli82016** [comment]
> > Is there a doc how to create a zone with an actual hardware deployment? Thanks.
> 
> Internally in GDCE we have several zones created for testing/development purposes, which I can use to run these acceptance tests. However I'm not sure it's a good idea to expose the details for them in a public GitHub repo. Alternatively, a GDCE zone can be purchased and provisioned via [these steps](https://cloud.google.com/distributed-cloud/edge/latest/docs/order).
> 
> Currently it is not possible to run any kind of E2E test for the edgenetwork API without owning some GDCE hardware, either provided by Google internally or purchased as a customer. To me, neither of these options seem to make sense for an open-source project.
> 
> I can confirm these tests are passing when I run them on a test zone. Would it be okay to disable these tests in the CI, and leave them as manual-run only?

You can remove these tests by adding ` skip_test: true`.  Example: https://github.com/GoogleCloudPlatform/magic-modules/blob/98a2f7047061f44e601405ecd31cdbf76f13f5a5/mmv1/products/iap/Brand.yaml#L55

There is no way to leave them as manual-run only.

**Review by zli82016 — APPROVED** [review]

**Comment by zli82016** [comment]
LGTM. Thanks.

**Pull request merged**